network_provisioning/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+if (NOT DEFINED ENV{IDF_PATH})
+    message(FATAL_ERROR "Please set IDF_PATH to the path of esp-idf repo")
+endif(NOT DEFINED ENV{IDF_PATH})
+
+set(IDF_PATH $ENV{IDF_PATH})
```

Comment:
Any specific reason for this, I don't  think this should be required here?

Reply:
This component requires a private include directory of `${IDF_PATH}/components/protocomm/proto-c`. So I set `IDF_PATH` here.

Reply:
The correct way to get `protocomm` component directory is like this:
```cmake
idf_component_get_property(protocomm_dir protocomm COMPONENT_DIR)
```
(we can't rely on the absolute path of the component under IDF_PATH, since the component may be overridden in the project)

Reply:
Used `idf_component_get_property` instead now